pennylane/workflow/execution.py

```diff
@@ -47,7 +47,7 @@
 logger = logging.getLogger(__name__)
 logger.addHandler(logging.NullHandler())
 
-device_type = Union[qml.Device, "qml.devices.Device"]
+SupportedDeviceAPIs = Union["qml.devices.LegacyDevice", "qml.devices.Device"]

```

Comment:
Since it's used a couple of times, shouldn't we place it on the `pennylane.typing.py`file?

Reply:
No this will be another instance of painful circular imports

Reply:
I see. Circular imports are the worst 😢

Reply:
Note: maybe using `__future__` module to import type hints would be a feasible solution.